minifi/minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/configuration/ConfigurationChangeCoordinator.java

```diff
@@ -99,16 +84,43 @@ public Collection<ListenerHandleResult> notifyListeners(ByteBuffer newConfig) {
                 result = new ListenerHandleResult(listener, ex);
             }
             listenerHandleResults.add(result);
-            logger.info("Listener notification result:" + result.toString());
+            LOGGER.info("Listener notification result: {}", result);
         }
         return listenerHandleResults;
     }
 
 
     @Override
-    public void close() throws IOException {
-        for (ChangeIngestor changeIngestor : changeIngestors) {
-            changeIngestor.close();
+    public void close() {
+        try {
+            for (ChangeIngestor changeIngestor : changeIngestors) {
+                changeIngestor.close();
+            }
+            changeIngestors.clear();
+        } catch (IOException e) {
+            LOGGER.warn("Could not successfully stop notifiers", e);
+        }
+    }
+
+    private void initialize() {
+        close();
+        // cleanup previously initialized ingestors
+        String ingestorsCsv = bootstrapProperties.getProperty(NOTIFIER_INGESTORS_KEY);
+
+        if (ingestorsCsv != null && !ingestorsCsv.isEmpty()) {
+            for (String ingestorClassname : ingestorsCsv.split(",")) {
+                ingestorClassname = ingestorClassname.trim();
+                try {
+                    Class<?> ingestorClass = Class.forName(ingestorClassname);
+                    ChangeIngestor changeIngestor = (ChangeIngestor) ingestorClass.newInstance();
+                    changeIngestor.initialize(bootstrapProperties, runMiNiFi, this);
+                    changeIngestors.add(changeIngestor);
+                    LOGGER.info("Initialized ingestor: {}", ingestorClassname);
+                } catch (Exception e) {
+                    LOGGER.error("Issue instantiating {} ingestor", ingestorClassname);
+                    LOGGER.error("Exception", e);
```

Comment:
Recommend consolidating these logs to a single invocation.
```suggestion
                    LOGGER.error("Instantiating [{}] ingestor failed", ingestorClassname, e);
```